### PR TITLE
BH-6503: handle applyPromotion 226 status code

### PR DESCRIPTION
**Pull request by kaleongchan**

### CODE REVIEW

**Review by trigger202 — DISMISSED** [review]

File: tests/Promotion/PromotionApiTest.php [hunk 1]
```diff
@@ -124,22 +124,27 @@ public function testApplyPromotionReturnsPromotionCode()
...
             json_encode($this->application->toArray())
         )->willReturn($this->response);
 
-        $this->response->expects($this->once())
+        $this->response->expects($this->exactly(2))
             ->method('getStatusCode')
-            ->willReturn(StatusCodeInterface::STATUS_CREATED);
+            ->willReturnOnConsecutiveCalls(
+                StatusCodeInterface::STATUS_CREATED,
+                StatusCodeInterface::STATUS_IM_USED,
```
Comment (afareh) [inline tests/Promotion/PromotionApiTest.php:136]: I think the trailing comma is making phpcs php7.1 fail

Comment (darkbluesun) [reply tests/Promotion/PromotionApiTest.php:136]: yep, can't have a trailing comma in a function call in php7.1

**Review by darkbluesun — DISMISSED** [review]
So this is new to me. https://httpstatuses.com/226
> The server has fulfilled a GET request for the resource, and the response is a representation of the result of one or more instance-manipulations applied to the current instance.

It doesn't say anything about POST requests, so I'm a bit confused.
I assume you've already had the discussion about implementing this in the promo ms and this is just the api sdk addition, so let me comment on that instead.
In that case, the changes are great! Just remove that comma for backwards compatibility.

**Review by DannyCui — DISMISSED** [review]

**Comment by kaleongchan** [comment]
> So this is new to me. https://httpstatuses.com/226
> 
> > The server has fulfilled a GET request for the resource, and the response is a representation of the result of one or more instance-manipulations applied to the current instance.
> 
> It doesn't say anything about POST requests, so I'm a bit confused.
> I assume you've already had the discussion about implementing this in the promo ms and this is just the api sdk addition, so let me comment on that instead.
> In that case, the changes are great! Just remove that comma for backwards compatibility.

Thanks @darkbluesun we had another chat and decided to change it to 200 to avoid confusion.

**Review by DannyCui — DISMISSED** [review]

File: src/Promotion/PromotionApi.php [hunk 2]
```diff
@@ -16,7 +16,9 @@ class PromotionApi extends AbstractApi
 
     /**
      * Tries apply promotion to application if its applicable for vendor and product type
-     * 202 status code means it not applicable
+     * 201 status code means it's applied
+     * 204 status code means it not applicable
+     * 226 status code means it's already applied
```
Comment (darkbluesun) [inline src/Promotion/PromotionApi.php:21]: should we change this then?

**Review by darkbluesun — APPROVED** [review]

**Review by darkbluesun — APPROVED** [review]
Very good! 
Conceptually, you could have used a dataprovider in the test rather than doing two requests in one request, but I don't mind.

**Review by DannyCui — APPROVED** [review]

**Pull request merged**